File: SearchSample/Objective-C/SearchSample/CategoryViewController.m

//
//  CategoryViewController.m
//  SearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "CategoryViewController.h"

@implementation CategoryViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [NTCategoryService executeAsync:^(NTCategoryResultSet * resultSet, NSError * error) {
        if (!error) {
            
            dispatch_async(dispatch_get_main_queue(), ^{
                _categories = resultSet.results;
                [_tableView reloadData];
            });
        }
    }];
    
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"categorytoResultSegue"]) {
        ResultViewController *resultViewController = segue.destinationViewController;
        [resultViewController searchByCategory:sender];
        
    }
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NTCategoryResult *category = _categories[indexPath.row];
    [self performSegueWithIdentifier:@"categorytoResultSegue" sender:category.code];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    NTCategoryResult *category = _categories[indexPath.row];
    cell.textLabel.text = category.name_L;
    
    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _categories != nil ? _categories.count : 0;
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

@end

File: SearchSample/Swift/SearchSample/CategoryViewController.swift
//
//  CategoryViewcontroller.swift
//  SearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

import UIKit
import NOSTRASDK

class CategoryViewController: UIViewController {
    
    @IBOutlet weak var tableView: UITableView!

    var categories: [NTCategoryResult]! = nil;
    
    override func viewDidLoad() {

        NTCategoryService.executeAsync { (resultSet, error) in
            DispatchQueue.main.async(execute: {
                
                if error == nil {
                    self.categories = resultSet?.results!;
                }
                else {
                    print("error \(error?.description)");
                }
                self.tableView.reloadData();
                
            });
            
        }
        
    
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "categorytoResultSegue" {
            let resultViewController = segue.destination as! ResultViewController;
            
            resultViewController.searchByCategory(sender as? String);
        }
    }

    func tableView(_ tableView: UITableView, didSelectRowAtIndexPath indexPath: IndexPath) {
        let category = categories[indexPath.row];
        self.performSegue(withIdentifier: "categorytoResultSegue", sender: category.code);
    }
    
    
    func tableView(_ tableView: UITableView, cellForRowAtIndexPath indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell");
        let category = categories[indexPath.row];
        
        cell?.textLabel?.text = category.name_E;
        
        return cell!;
        
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return categories != nil ? (categories?.count)! : 0;
    }
    
    func numberOfSectionsInTableView(_ tableView: UITableView) -> Int {
        return 1
    }

    
}
